implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -53,23 +53,20 @@ public JwtContext parse(final String token, final JWTAuthContextInfo authContext
             JwtConsumerBuilder builder = new JwtConsumerBuilder()
                     .setRequireExpirationTime();
 
-            if (authContextInfo.getWhitelistAlgorithms().isEmpty()) {
-                builder.setJwsAlgorithmConstraints(
-                        new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
-                                AlgorithmIdentifiers.RSA_USING_SHA256));
-            } else {
+            if (!authContextInfo.getWhitelistAlgorithms().isEmpty()) {
```

Comment:
@aaronanderson white list should not be empty at this stage. Yes we have deprecated the public property, but internally it should be set at this point to the current signature algorithm